resources/schema.gql

```diff
@@ -610,6 +651,64 @@ type FlowConfigurationBatching {
 
 union FlowConfigurationSchedule = TimeDelta | CronExpression
 
+type FlowConnection {
+	"""
+	A shorthand for `edges { node { ... } }`
+	"""
+	nodes: [Flow!]!
+	"""
+	Approximate number of total nodes
+	"""
+	totalCount: Int!
+	"""
+	Page information
+	"""
+	pageInfo: PageBasedInfo!
+	edges: [FlowEdge!]!
+}
+
+union FlowDescription = FlowDescriptionDataset | FlowDescriptionSystem
+
+type FlowDescriptionCompaction {
+	originalBlocksCount: Int!
+	resultingBlocksCount: Int
+}
+
+type FlowDescriptionDataset {
+	datasetId: DatasetID!
+	details: FlowDescriptionDatasetDetails!
+}
+
+union FlowDescriptionDatasetDetails = FlowDescriptionPollingIngest | FlowDescriptionPushIngest | FlowDescriptionExecuteQuery | FlowDescriptionCompaction
```

Comment:
Can we glue these two types into one? The union can require `datasetId` as a field.

Reply:
We are not allowed to have unions of unions in GraphQL, but I've utilized `flatten` attribute, which makes a single union from nested unions.

`union FlowDescription = FlowDescriptionDatasetPollingIngest | FlowDescriptionDatasetPushIngest | FlowDescriptionDatasetExecuteQuery | FlowDescriptionDatasetCompaction | FlowDescriptionSystem`

Unions may not have fields, unlike interfaces, but I guess we can live with a flattened 1 property.